src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,247 @@
+  }
+
+  private ngUnsubscribe = new Subject();
+
+  private _defaultCountryCode: string;
+
+  private _disabled: boolean;
+
+  private _modelValue: string;
+
+  public constructor(
+    private adapterService: SkyPhoneFieldAdapterService,
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private injector: Injector
+  ) { }
+
+  public ngOnInit(): void {
+    this.adapterService.addElementClass(this.elRef.nativeElement, 'sky-form-control');
+    if (this.defaultCountry) {
+      this.skyPhoneFieldComponent.selectCountry(this.defaultCountry);
+    }
+    this.adapterService.setElementPlaceholder(this.elRef.nativeElement,
+      this.skyPhoneFieldComponent.selectedCountry.placeholder);
+
+    this.adapterService.setAriaLabel(this.elRef.nativeElement);
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+    this.selectedCountryChange.complete();
+  }
+
+  public ngAfterViewInit(): void {
+    this.skyPhoneFieldComponent.selectedCountryChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((country: SkyCountryData) => {
+        // Write the value again to cause validation to refire
+        this.writeValue(this.modelValue);
+        this.adapterService.setElementPlaceholder(this.elRef.nativeElement, country.placeholder);
+        this.selectedCountryChange.emit(country);
+      });
+    // This is needed to address a bug in Angular 4, where the value is not changed on the view.
+    // See: https://github.com/angular/angular/issues/13792
+    const control = (<NgControl>this.injector.get(NgControl)).control as FormControl;
+    /* istanbul ignore else */
+    if (control && this.modelValue) {
+      control.setValue(this.modelValue, { emitEvent: false });
+      /* istanbul ignore else */
+      if (this.changeDetector) {
+        this.changeDetector.detectChanges();
+      }
+    }
+  }
+
+  /**
+   * Writes the new value for reactive forms on change events on the input element
+   * @param event The change event that was received
+   */
+  @HostListener('change', ['$event'])
+  public onChange(event: any): void {
+    this.writeValue(event.target.value);
+  }
+
+  /**
+   * Marks reactive form controls as touched on input blur events
+   */
+  @HostListener('blur')
+  public onBlur(): void {
+    this._onTouched();
+  }
+
+  /**
+   * Writes the new value for reactive forms
+   * @param value The new value for the input
+   */
+  public writeValue(value: any): void {
+    this.modelValue = value;
+  }
+
+  public registerOnChange(fn: (value: any) => any): void { this._onChange = fn; }
+
+  public registerOnTouched(fn: () => any): void { this._onTouched = fn; }
+
+  public registerOnValidatorChange(fn: () => void): void { this._validatorChange = fn; }
+
+  /**
+   * Sets the disabled state on the input
+   * @param isDisabled the new value of the input's disabled state
+   */
+  public setDisabledState(isDisabled: boolean): void {
```

Comment:
I don't see this method being used anywhere. Is it necessary? It doesn't seem needed for your `ControlValueAccessor` implementation either.

Reply:
It's part of the `ControlValueAccessor`, see: https://angular.io/api/forms/ControlValueAccessor#setDisabledState

Reply:
Weird...i didn't get any IDE warnings when I removed it. Ok, just ignore this comment!

Reply:
@Blackbaud-AlexKingman just for full clarity the `ControlValueAccessor` uses this method under the hood and it is needed to allow for consumers with reactive forms to be able to call `disable` and `enable` on their form control. It is an optional method on the `ControlValueAccessor` which is why it was not throwing any warnings but removing it would cause `disable` and `enable` calls on the form to have no affect.